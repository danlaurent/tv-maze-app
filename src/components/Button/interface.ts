export interface IButton {
  text: string;
  onPress: () => void;
  testID?: string;
}
