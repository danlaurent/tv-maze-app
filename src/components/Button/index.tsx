import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { LIGHT_GRAY } from '../../constants/styles/colors';
import { IButton } from './interface';

const styles = StyleSheet.create({
  button: {
    height: 48,
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderRadius: 14,
  },
  text: {
    fontSize: 14,
    textAlign: 'center',
    color: LIGHT_GRAY,
  },
});

const Button = ({ text, onPress, testID }: IButton) => (
  <TouchableOpacity testID={testID} style={styles.button} onPress={onPress}>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

export default Button;
