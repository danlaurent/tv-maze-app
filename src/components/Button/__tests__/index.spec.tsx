import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import Button from '..';

const onPressMock = jest.fn();

describe('Button', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <Button text='Test' onPress={() => onPressMock()} testID='testButton' />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the button is pressed', () => {
    beforeEach(async () => {
      await fireEvent.press(tree.getByTestId('testButton'));
    });

    it('fires the onPress action', () => {
      expect(onPressMock).toBeCalledTimes(1);
    });
  });
});
