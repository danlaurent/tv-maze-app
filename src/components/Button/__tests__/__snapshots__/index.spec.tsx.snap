// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 14,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
  testID="testButton"
>
  <Text
    style={
      Object {
        "color": "#c6c7d2",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    Test
  </Text>
</View>
`;
